File: src/main/java/me/tigermouthbear/nebulous/modifiers/AllCapsModifier.java
package me.tigermouthbear.nebulous.modifiers;

import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.AbstractInsnNode;
import org.objectweb.asm.tree.ClassNode;
import org.objectweb.asm.tree.LdcInsnNode;
import org.objectweb.asm.tree.MethodNode;

public class AllCapsModifier extends Modifier
{
	public void modify()
	{
		for(ClassNode cn: getClassMap().values())
		{
			for(MethodNode mn: cn.methods)
			{
				// Iterate instructions in method
				for(AbstractInsnNode ain: mn.instructions.toArray())
				{
					// If the instruction is loading a constant value
					if(ain.getOpcode() == Opcodes.LDC)
					{
						// Cast current instruction to Ldc
						// If the constant is a string then capitalize it.
						LdcInsnNode ldc = (LdcInsnNode) ain;
						if(ldc.cst instanceof String)
						{
							ldc.cst = ldc.cst.toString().toUpperCase();
						}
					}
				}
			}
		}
	}
}

File: src/main/java/me/tigermouthbear/nebulous/modifiers/AllCapsModifier.kt
package me.tigermouthbear.nebulous.modifiers

import org.objectweb.asm.Opcodes
import org.objectweb.asm.tree.LdcInsnNode

class AllCapsModifier : Modifier() {
    override fun modify() {
        for (cn in classMap.values) {
            for (mn in cn.methods) { // Iterate instructions in method
                for (ain in mn.instructions.toArray()) { // If the instruction is loading a constant value
                    if (ain.opcode == Opcodes.LDC) { // Cast current instruction to Ldc
// If the constant is a string then capitalize it.
                        val ldc = ain as LdcInsnNode
                        if (ldc.cst is String) {
                            ldc.cst = ldc.cst.toString().toUpperCase()
                        }
                    }
                }
            }
        }
    }
}